test/new-e2e/pkg/utils/e2e/client/agent_client.go

```diff
@@ -169,3 +171,97 @@ func curlCommand(endpoint string, authToken string) string {
 		endpoint,
 	)
 }
+
+func waitForReadyTimeout(t *testing.T, host *Host, commandRunner *agentCommandRunner, timeout time.Duration) error {
+	err := commandRunner.waitForReadyTimeout(timeout)
+
+	if err != nil {
+		// Propagate the original error if we have another error here
+		localErr := generateAndDownloadFlare(t, commandRunner, host)
+
+		if localErr != nil {
+			t.Errorf("Could not generate and get a flare: %v", localErr)
+		}
+	}
+
+	return err
+}
+
+func generateAndDownloadFlare(t *testing.T, commandRunner *agentCommandRunner, host *Host) error {
+	profile := runner.GetProfile()
+	outputDir, err := profile.GetOutputDir()
+	flareFound := false
+
+	if err != nil {
+		return fmt.Errorf("could not get output directory: %v", err)
+	}
+
+	_, err = commandRunner.FlareWithError(agentclient.WithArgs([]string{"--email", "e2e@test.com", "--send", "--local"}))
+	if err != nil {
+		t.Errorf("Error while generating the flare: %v.", err)
+		// Do not return now, the flare may be generated locally but was not uploaded because there's no fake intake
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do look for something to sincerely praise. -->
**👏 praise**